common/searchattribute/system_provider.go

```diff
@@ -32,52 +32,25 @@ type (
 	SystemProvider struct{}
 )
 
-var (
-	// system are default internal system search attributes which are shown to the users.
-	// Note: NamespaceID is not included.
-	system = map[string]enumspb.IndexedValueType{
-		WorkflowID:      enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		RunID:           enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		WorkflowType:    enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		StartTime:       enumspb.INDEXED_VALUE_TYPE_INT,
-		ExecutionTime:   enumspb.INDEXED_VALUE_TYPE_INT,
-		CloseTime:       enumspb.INDEXED_VALUE_TYPE_INT,
-		ExecutionStatus: enumspb.INDEXED_VALUE_TYPE_INT,
-		TaskQueue:       enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-
-		TemporalChangeVersion: enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		BinaryChecksums:       enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		CustomNamespace:       enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-		Operator:              enumspb.INDEXED_VALUE_TYPE_KEYWORD,
-	}
-)
-
 func NewSystemProvider() *SystemProvider {
 	return &SystemProvider{}
 }
 
-func (s *SystemProvider) GetSearchAttributes(_ string, _ bool) (map[string]enumspb.IndexedValueType, error) {
-	return system, nil
-}
-
-func GetSystem() map[string]enumspb.IndexedValueType {
-	return system
+func (s *SystemProvider) GetSearchAttributes(_ string, _ bool) (NameTypeMap, error) {
+	return NameTypeMap{}, nil
 }
 
-// FilterCustom returns new search attributes map with only custom search attributes in it.
-func FilterCustom(searchAttributes map[string]enumspb.IndexedValueType) map[string]enumspb.IndexedValueType {
+// FilterCustomOnly returns new search attributes map with only custom search attributes in it.
+// TODO: Remove this after 1.10.0 release
+func FilterCustomOnly(searchAttributes map[string]enumspb.IndexedValueType) map[string]enumspb.IndexedValueType {
 	customSearchAttributes := map[string]enumspb.IndexedValueType{}
 	for saName, saType := range searchAttributes {
+		if saName == NamespaceID {
```

Comment:
why special treating the namespace ID? or should other system search attr be excluded here?

Reply:
Yes, `NamespaceId` is different. It should be hidden from user and I will add check for query parser later.